mmaction/datasets/pipelines/augmentations.py

```diff
@@ -868,6 +870,406 @@ def __repr__(self):
         return repr_str
 
 
+@PIPELINES.register_module()
+class CuboidCrop:
+    """Crop cuboid candidates and randomly sample one to form tubelet.
+
+    Required keys in results are "img_shape", "gt_bboxes" and "imgs", added or
+    modified keys are "imgs", "img_shape" and "gt_bboxes".
+
+    Args:
+        cuboid_settings (list[dict]): Setting dict for cropping cuboid
+            candidates. Each dict may contain keys like "max_trials",
+            "max_sample", "sampler" ("sample" is also a dict which contains
+            keys like "min_scale", "max_scale", "min_aspect", "max_aspect"),
+            "constraints" ("constraints" is also a dict which contains keys
+            like "min_jaccard_overlap", "max_jaccard_overlap").
+    """
+
+    def __init__(self, cuboid_settings):
+        self.cuboid_settings = cuboid_settings
+
+    def sample_cuboids(self, tubes, img_h, img_w):
+        tubes = sum(tubes.values(), [])
+        sampled_cuboids = []
+        for cuboid_setting in self.cuboid_settings:
+            trial = 0
+            sample = 0
+            max_trials = cuboid_setting['max_trials']
+            max_sample = cuboid_setting['max_sample']
+            sampler = cuboid_setting['sampler']
+
+            min_scale = sampler.get('min_scale', 1)
+            max_scale = sampler.get('max_scale', 1)
+            min_aspect = sampler.get('min_aspect', 1)
+            max_aspect = sampler.get('max_aspect', 1)
+
+            while trial < max_trials and sample < max_sample:
+                # sample a normalized box
+                scale = np.random.uniform(min_scale, max_scale)
+                aspect = np.random.uniform(min_aspect, max_aspect)
+
+                w = scale * np.sqrt(aspect)
+                h = scale / np.sqrt(aspect)
+                if w > 1 or h > 1:
+                    continue
+                x = np.random.uniform(0, 1 - w)
+                y = np.random.uniform(0, 1 - h)
+
+                # rescale the box
+                sampled_cuboid = np.array(
+                    [x * img_w, y * img_h, (x + w) * img_w, (y + h) * img_h],
+                    dtype=np.float32)
+                # check constraint
+                trial += 1
+                if 'constraints' not in cuboid_setting:
+                    sampled_cuboids.append(sampled_cuboid)
+                    sample += 1
+                    continue
+
+                contraints = cuboid_setting['constraints']
+                min_jaccard_overlap = contraints.get('min_jaccard_overlap',
+                                                     np.inf)
```

Comment:
double check inf or -inf

Reply:
it should be inf, the original code is below
![image](https://user-images.githubusercontent.com/33949370/100617543-a34e8180-3355-11eb-97e8-3e843bc8be5d.png)